app/src/main/java/org/thoughtcrime/securesms/onboarding/messagenotifications/MessageNotificationsActivity.kt

```diff
@@ -39,13 +45,15 @@ class MessageNotificationsActivity : BaseActionBarActivity() {
     }
 
     private fun register() {
-        TextSecurePreferences.setPushEnabled(this, viewModel.stateFlow.value.pushEnabled)
+        prefs.setPushEnabled(viewModel.stateFlow.value.pushEnabled)
         ApplicationContext.getInstance(this).startPollingIfNeeded()
         pushRegistry.refresh(true)
-        Intent(this, HomeActivity::class.java).apply {
-            flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
-            putExtra(HomeActivity.FROM_ONBOARDING, true)
-        }.also(::startActivity)
+
+        when {
+            prefs.getHasViewedSeed() && !prefs.getConfigurationMessageSynced() -> start<LoadingActivity>()
+            prefs.getProfileName() != null -> startHomeActivity()
+            else -> startHomeActivity()
```

Comment:
The last two branch of the `when` do the same thing, is that normal?

Reply:
cleaned up.